src/ft/mod.rs

```diff
@@ -481,17 +480,22 @@ impl FreeTypeRasterizer {
             buf,
         };
 
-        if face.colored {
-            let fixup_factor = if let Some(pixelsize_fixup_factor) = face.pixelsize_fixup_factor {
-                pixelsize_fixup_factor
-            } else {
-                // Fallback if user has bitmap scaling disabled.
-                let metrics = face.ft_face.size_metrics().ok_or(Error::MissingSizeMetrics)?;
-                f64::from(pixelsize) / f64::from(metrics.y_ppem)
-            };
-            Ok(downsample_bitmap(rasterized_glyph, fixup_factor))
-        } else {
-            Ok(rasterized_glyph)
+        match (is_missing_glyph, face.colored) {
```

Comment:
My mash these together when all you do for a missing glyph is just return early anyways? That seems unnecessarily complicated, wouldn't it be cleaner to just have a separate check for that which returns right before this?

Or at least have an `if/elseif/else` block, rather than a 'match everything at the same time' approach. That's usually easier to follow and probably the best idea here when we're dealing only with booleans anyways.